src/storage/store/column_chunk.cpp

```diff
@@ -0,0 +1,343 @@
+#include "storage/store/column_chunk.h"
+
+#include "storage/copier/table_copy_utils.h"
+#include "storage/storage_structure/storage_structure_utils.h"
+#include "storage/store/struct_column_chunk.h"
+#include "storage/store/var_sized_column_chunk.h"
+
+using namespace kuzu::common;
+using namespace kuzu::transaction;
+
+namespace kuzu {
+namespace storage {
+
+ColumnChunk::ColumnChunk(LogicalType dataType, CopyDescription* copyDescription, bool hasNullChunk)
+    : ColumnChunk{
+          std::move(dataType), StorageConstants::NODE_GROUP_SIZE, copyDescription, hasNullChunk} {}
+
+ColumnChunk::ColumnChunk(
+    LogicalType dataType, offset_t numValues, CopyDescription* copyDescription, bool hasNullChunk)
+    : dataType{std::move(dataType)}, numBytesPerValue{getDataTypeSizeInChunk(this->dataType)},
+      numBytes{numBytesPerValue * numValues}, copyDescription{copyDescription} {
+    buffer = std::make_unique<uint8_t[]>(numBytes);
+    if (hasNullChunk) {
+        nullChunk = std::make_unique<NullColumnChunk>();
+    }
+}
+
+void ColumnChunk::resetToEmpty() {
+    if (nullChunk) {
+        nullChunk->resetNullBuffer();
+    }
+}
+
+void ColumnChunk::appendVector(
+    ValueVector* vector, offset_t startPosInChunk, uint32_t numValuesToAppend) {
+    assert(vector->dataType.getLogicalTypeID() == LogicalTypeID::ARROW_COLUMN);
+    auto array = ArrowColumnVector::getArrowColumn(vector).get();
+    appendArray(array, startPosInChunk, numValuesToAppend);
+}
+
+void ColumnChunk::appendColumnChunk(ColumnChunk* other, offset_t startPosInOtherChunk,
+    offset_t startPosInChunk, uint32_t numValuesToAppend) {
+    if (nullChunk) {
+        nullChunk->appendColumnChunk(
+            other->nullChunk.get(), startPosInOtherChunk, startPosInChunk, numValuesToAppend);
+    }
+    memcpy(buffer.get() + startPosInChunk * numBytesPerValue,
+        other->buffer.get() + startPosInOtherChunk * numBytesPerValue,
+        numValuesToAppend * numBytesPerValue);
+}
+
+void ColumnChunk::appendArray(
```

Comment:
maybe call all of these append() if you can (but I believe in C++ you could not override the functions with different arguments). If not you can call appendArrowArray. Array is a very general term.